sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -338,6 +338,10 @@ class RegexParser(pattern: String) {
         throw new RegexUnsupportedException("escape at end of string", Some(pos))
       case Some(ch) =>
         ch match {
+          case 'r' | 'n' | 'f' =>
```

Comment:
Why do we need this? Wouldn't it treat `\\n` as the character literal `\n`

Reply:
I ran into an exception something like "unsupported escape character 'r'". If you comment out this code and run the tests you should be able to reproduce it.

Reply:
See [my comment](https://github.com/NVIDIA/spark-rapids/pull/5610#discussion_r880892707). This case should be removed since it would incorrectly convert escaped characters to literals and throw the wrong error message.

Reply:
Agreed. I am working on removing this now.